### ISSUE TITLE
CWG2887 [diff.cpp03.expr] Should we list the behavioral change of `typeid` in C++11 due to N3055?

### ISSUE THREAD

**Issue by frederick-vs-ja** [cplusplus/draft#6861]
In the following example, the operand of `typeid` (an xvalue of a polymorphic class type) wasn't evaluated in C++98, but was made to evaluated since C++11 due to [N3055](https://wg21.link/n3055).
```C++
#include <typeinfo>
#include <cstdio>

struct B {
  virtual ~B() { std::puts("~B"); }
};

struct D : B {};

struct WrapB {
  B b;
};

struct WrapD {
  D d;
};

int main() {
  typeid(true ? WrapB().b : WrapD().d);
}
```

**Comment by t3nsor** [comment]
Issues like this need to be submitted to CWG so that we can review the example

**Comment by frederick-vs-ja** [comment]
@t3nsor Submitted an issue in the CWG repo.

**Comment by frederick-vs-ja** [comment]
Should be fixed by #7099 (84006114ce2ff8012a740509b409b324b153a17e).
